bin/syclcc

```diff
@@ -222,8 +226,62 @@ class source_file_processor:
 
 cpp_files = set([".cpp",".cxx",".c++",".cc",".c"])
 
-## Compiler backend using nvcc
-class cuda_compiler:
+## Compiler backend for CUDA using Clang
+class cuda_clang_compiler:
+  def __init__(self,config):
+    self.device_compiler = config.cuda_clang_compiler
+    self.config = config
+    self.is_clang_available = is_program_executable(self.device_compiler)
+
+  def includes_linking_stage(self, args):
+    for arg in args:
+      if (arg.startswith("-E") or
+          arg.startswith("-fsyntax-only") or
+          arg.startswith("-S") or
+          arg.startswith("-c")):
+        return False
+    return True
+
+  def run(self, args):
+    source_processor = source_file_processor(self.config, ".cu", args)
+    result = None
+    try:
+      transformed_args = []
+      for arg in args:
+        if is_source_file(arg):
+          transformed_args.append(source_processor.process_file(arg))
+        else:
+          transformed_args.append(arg)
+
+      compiler_args = ["--cuda-gpu-arch=sm_52",
```

Comment:
Won't this lead to clashes if the user wants to override the the arch and passes `--cuda-gpu-arch` himself? The `hip_compiler` only appends the GPU arch argument based on the content of  `HIPSYCL_GPU_ARCH`, if this environment variable is defined. Perhaps we can have a similar behavior here for the sake of consistency and only add `--cuda-gpu-arch=sm_52` if we don't have an arch defined in `HIPSYCL_GPU_ARCH`?

Reply:
Well the user-provided arguments are appended afterwards, so I thought they'd just overwrite the default. *But* I just realized that apparently [it is possible to pass multiple architectures at once](http://llvm.org/docs/CompileCudaWithLLVM.html#invoking-clang), and compile for all of them. So yes, this is not a good solution. Also I didn't realize you already had a `HIPSYCL_GPU_ARCH` in place, as `nvcc` doesn't use it. I will pull this out of `hip_compiler` into a general configuration option, is that okay with you?

Reply:
Perfect, that sounds like a good solution :)